cgk_calcium_tools/filehandler.py

```diff
@@ -361,28 +341,39 @@ def get_pair_filenames(self, operation: str) -> Tuple[list, list]:
         -------
         Tuple[list,list]
             pair of input and output file paths
-
         """
+
+        # Checks if operation is in the lsit of processing steps
         assert (
             operation in self.processing_steps
         ), f"operation must be in the list {self.processing_steps}, got: {operation}"
 
-        opi = self.processing_steps.index(operation)
+        # Copy the list of processing steps to a local variable
+        local_processing_steps = self.processing_steps
```

Comment:
self.processing_steps is a list (not an immutable as an int or str), then any change in local_processing_steps will affect self.processing_steps